llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -2291,9 +2291,10 @@ class VPWidenPHIRecipe : public VPSingleDefRecipe {
   SmallVector<VPBasicBlock *, 2> IncomingBlocks;
 
 public:
-  /// Create a new VPWidenPHIRecipe for \p Phi with start value \p Start.
-  VPWidenPHIRecipe(PHINode *Phi, VPValue *Start = nullptr)
-      : VPSingleDefRecipe(VPDef::VPWidenPHISC, ArrayRef<VPValue *>(), Phi) {
+  /// Create a new VPWidenPHIRecipe for \p Phi with start value \p Start and
+  /// debug location \p DL.
+  VPWidenPHIRecipe(PHINode *Phi, VPValue *Start = nullptr, DebugLoc DL = {})
+      : VPSingleDefRecipe(VPDef::VPWidenPHISC, ArrayRef<VPValue *>(), Phi, DL) {
```

Comment:
Why not pass debug info by this way? Could Phi be nullptr, or other reasons?
```suggestion
  /// Create a new VPWidenPHIRecipe for \p Phi with start value \p Start.
  VPWidenPHIRecipe(PHINode *Phi, VPValue *Start = nullptr)
      : VPSingleDefRecipe(VPDef::VPWidenPHISC, ArrayRef<VPValue *>(), Phi, Phi->getDebugLoc()) {
```

Reply:
I see, considering that Phi could potentially be nullptr is reasonable. 
But I have another question: Do we allow `phi->getDebugLoc()` and `DL` to be different? If not, should we add an assert in the constructor? If we do allow it, should we use `phi->getDebugLoc()` (if phi is not nullptr) or `DL`?
cc: @fhahn

Reply:
IMO it is better to pass the debug loc separately; the phi instruction is technically optionally